### PR TITLE
ENH: ibis.pandas.from_dataframe convenience function

### LINKED ISSUE #1155: ENH: .from_dataframe() convenience function
```
In [7]: import ibis

In [8]: df = pd.DataFrame({'A': [1, 2, 3]})

In [9]: t = ibis.pandas.connect({'df': df}).table('df')

In [10]: t.A+1
Out[10]: 
ref_0
PandasTable[table]
  name: df
  schema:
    A : int64

Add[int64*]
  left:
    A = Column[int64*] 'A' from table
      ref_0
  right:
    Literal[int8]
      1
```

would be nice to write this like
```
from ibis.pandas.api import from_dataframe
t = from_dataframe(pd.DataFrame(.....))
```

makes interop with pandas that much easier

### PR DESCRIPTION
**Pull request by jreback**
closes #1155

### CODE REVIEW

File: ibis/pandas/api.py [hunk 1]
```diff
@@ -7,3 +7,19 @@
 
 def connect(dictionary):
     return PandasClient(dictionary)
+
+
+def from_dataframe(df):
```
Comment (cpcloud) [inline ibis/pandas/api.py]: Let's also add a `client=None` parameter and a `name='df'` parameter.

Comment (jreback) [reply ibis/pandas/api.py]: done

File: ibis/pandas/api.py [hunk 2]
```diff
@@ -7,3 +7,24 @@
 
 def connect(dictionary):
     return PandasClient(dictionary)
+
+
+def from_dataframe(df, name='df', client=None):
```
Comment (cpcloud) [inline ibis/pandas/api.py]: Sorry, I don't think I was clear here. What I meant was an *instance* of `PandasClient`, so what you want is something like:

```python
if client is None:
    return connect({name: df}).table(name)
client.dictionary[name] = df
return client.table(name)
```

Comment (jreback) [reply ibis/pandas/api.py]: pushed as you suggested. this *does* mutate the client, is that ok?

Comment (cpcloud) [reply ibis/pandas/api.py]: Yeah, I think that's fine. Can you add some documentation to the `client` parameter to this effect?

File: ibis/pandas/api.py [hunk 3]
```diff
@@ -9,17 +9,23 @@ def connect(dictionary):
     return PandasClient(dictionary)
 
 
-def from_dataframe(df):
+def from_dataframe(df, name='df', client=None):
     """
     convenience function to construct an ibis table
     from a DataFrame
```
Comment (cpcloud) [inline ibis/pandas/api.py]: Can you mark this as experimental? I'm not sure we'll want to keep this particular version around indefinitely. Let's see what shakes out after a few releases with people using this.

**Comment by jreback** [comment]
updated

**Comment by jreback** [comment]
@cpcloud ping

**Comment by cpcloud** [comment]
This looks good. Will merge when I get in.

**Comment by jreback** [comment]
ping

**Comment by jreback** [comment]
rebased

**Comment by cpcloud** [comment]
@jreback can you rebase one more time, then i will merge

**Comment by jreback** [comment]
rebased

**Comment by cpcloud** [comment]
thanks @jreback

**Pull request closed without merging**